numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -4567,6 +4567,11 @@ PyMODINIT_FUNC init_multiarray_umath(void) {
     PyObject *m, *d, *s;
     PyObject *c_api;
 
+    /* Initialize CPU features */
+    if (npy_cpu_init() < 0) {
+        return RETVAL(NULL);
+    }
+
```

Comment:
This should be closer to the `s = npy_cpu_features_dict();` call below

Reply:
`npy_cpu_init()` must be called before `InitOperators()` since `npy_cpu_have` is used by umath generator

Reply:
ok, then move it closer to that call. It does not need to be before the module creation code.

Reply:
currently, it doesn't return any errors but within the second pull request it will return the following errors:
- the validation errors of disabling CPU features via the environment variable.
- the validation errors of the baseline feature availability.

so it makes more sense to call it before ushering the module.

Reply:
leaving this for now as is